test/sh/failpods.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+#
+# Copyright (c) 2022. Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+#
+
+namespace=""
+maxPods=10
+
+for param in $*; do
+  case $param in
+  "--ns")
+    shift
+    namespace=$1
+    shift
+    ;;
+  "--timeoutseconds")
+    shift
+    timeoutseconds=$1
+    shift
+    ;;
+  "--help")
+    shift
+    echo "parameters: --ns driver_namespace [--timeoutseconds value]"
+    exit
+    ;;
+  esac
+done
+
+[ "$namespace" = "" ] && echo "Required argument: --ns driver_namespace --timeoutseconds value" && exit 2
+
+# nodeList returns a list of nodes
+nodeList() {
+	kubectl get nodes -A | grep -v -E 'mast.r|control-plane'  | grep -v NAME | awk '{ print $1 }'
+}
+
+# getNumOfPods returns the number of pods on a node $1 for a specific namespace
+getNumOfPods() {
+	# Add an extra space on match string to differentiate worker-1 from worker-10
+	kubectl get pods -A -o wide | grep "$1 " | grep $namespace | wc -l 
+}
+
+# getWorker returns the first worker node that we are targeting
+getWorker(){
+    kubectl get nodes -A | grep -v -E 'mast.r|control-plane'  | grep -v NAME | awk 'NR==1{ print $1 }'
+}
+
+# getRunningPods returns the names of the running pods that are on the targeted worker node for a specific namespace
+getRunningPods() {
+    node=$(getWorker)
+    # kubectl get pods -A -o wide | grep $node | grep $namespace | grep Running | awk '{ print $2 }'
+    kubectl get pods -A -o wide | grep $node | grep $namespace | grep -v 'controller' | awk '{ print $2 }'
+}
+
+# getDriverImage returns the initial driver image before its patched
+getDriverImage() {
+    ns=$namespace
+    pods=$(getRunningPods)
+    for pod in $pods; do
+        kubectl get pod $pod -n $ns -o custom-columns=IMAGE:.spec.containers[1].image | awk 'FNR == 2 {print}'
+    done
+}
+
+# failPodsInNS will fail the pods for a specific namespace by patching it with an unknown driver image
+failPodsInNS() {
+    ns=$namespace
+    pods=$(getRunningPods)
```

Comment:
Since you want pods in a specific namespace, shouldn't getRunningPods take an argument of the namespace to filter by? With many of our tests in podmon, the pod names are the same, e.g. pmtv1/podmontest-0, pmtv2/podmontest-0, ... I don't know if you will have that problem or not- your discretion.

Reply:
Hi @rbo54, the getRunningPods function does take in the namespace as an argument

Reply:
My mistake. Thanks.